keyvalue.go

```diff
@@ -38,4 +38,8 @@ type KeyValue interface {
 	// allocation-heavy. Consider implementing the LogMarshaler interface instead.
 	AddObject(key string, value interface{}) error
 	AddString(key, value string)
+
+	// Slices
+	AddInts(key string, value []int)
+	AddStrings(key string, value []string)
```

Comment:
nit: can we put `AddInts` immediately after `AddInt`, and likewise for `AddStrings`? I think that the type signatures are enough documentation - there's no need for an additional comment.